ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java

```diff
@@ -194,6 +197,28 @@ void compact(CompactionDetails details, byte[] bundleReadBuffer) throws IOExcept
     resumeCompaction(bundleReadBuffer);
   }
 
+  /**
+   * Converts the set of {@link Container}s to the set of accountId & ContainerId pairs for easy detection.
+   */
+  private void getAllContainersByStatus(EnumSet<Container.ContainerStatus> containerStatusSet) {
+    Set<Pair<Short, Short>> selectedContainerPairSet = new HashSet<>();
+    if (accountService != null) {
+      for (Container.ContainerStatus containerStatus : containerStatusSet) {
+        switch (containerStatus) {
+          case INACTIVE:
+          case DELETE_IN_PROGRESS:
+            accountService.getContainersByStatus(containerStatus).forEach((container) -> {
+              selectedContainers.add(new Pair<>(container.getParentAccountId(), container.getId()));
+            });
+            break;
+          default:
+            logger.error("Invalid container status: {}", containerStatus);
+        }
```

Comment:
why adding a switch case here? In this method we shouldn't really care about the container status, rather than just return all the container with the given status.

Reply:
It's due to I need to separately handle the INACTIVE and DELETE_IN_PROGRESS status for the future. To support the ACLs during compaction, I need to filter out the DELETE_IN_PROGRESS container which not qualify with the retention time, so only all Inactive containers and qualified Delete_in_progress containers can be remove during that compaction cycle. Does that make sense to you?

Reply:
it's weird that in line 251 we already specified the status set, but here we have to deal with it again. If this is the case, I encourage you to just ignore status in line 251, by changing this method to something like `getAllFilteredContainers`.

Reply:
Updated.